### PR TITLE
add lucide icons as a candidate for core icon library

### PR DESCRIPTION
**Pull request by Lendemor**
`size`, `strokeWidth` and `color` can be customized like in the following example:

```python
import reflex as rx

rx.lucide.icon(tag="moon", size=24, strokeWidth="1"),
rx.lucide.icon(tag="moon", size=48, strokeWidth="2"),
rx.lucide.icon(tag="moon", size=24, strokeWidth="2", color="red"),
rx.lucide.icon(tag="moon", size=48, strokeWidth="2", color="blue"),
```

image

### CODE REVIEW

File: reflex/components/lucide/icon.py
Comment (Alek99) [inline reflex/components/lucide/icon.py:1]: Hard  pin version here

**Review by Alek99 — CHANGES_REQUESTED** [review]
### Comments from discord 
he only comment I have with this is that if they are using the theme provider I would like this to default to using it so it can look good in a button like so https://www.radix-ui.com/themes/docs/components/button
We may need to make a method that can check if a theme is being used
We can also get rid of radix icons as we only need one icon lib

File: reflex/components/lucide/icon.py
Comment (Alek99) [reply reflex/components/lucide/icon.py:1]: This is the only one needing to be addressed now

**Review by picklelo — APPROVED** [review]
awesome

**Pull request merged**